pkg/asset/machines/aws/aws.go

```diff
@@ -0,0 +1,93 @@
+// Package aws generates Machine objects for aws.
+package aws
+
+import (
+	"text/template"
+
+	"github.com/openshift/installer/pkg/types"
+)
+
+// Config is used to generate the machine.
+type Config struct {
+	ClusterName string
+	Replicas    int64
+	Tags        map[string]string
+	Region      string
+	Machine     types.AWSMachinePoolPlatform
+}
+
+// WorkerMachineSetTmpl is template for worker machineset.
+var WorkerMachineSetTmpl = template.Must(template.New("aws-worker-machineset").Parse(`
+apiVersion: cluster.k8s.io/v1alpha1
+kind: MachineSet
+metadata:
+  name: {{.ClusterName}}-worker-0
```

Comment:
What's up the `-0` suffix on these? This will be copied into `generateName` by the controller, so the nodes will end up with names like `openshift-worker-0-abdef`.

Reply:
> What's up the -0 suffix on these

We there are going to have multiple machinesets (in aws per az).

> so the nodes will end up with names like openshift-worker-0-abdef

I think the node names is dictated by the machine object name. it will be whatever kubelet decides. on aws it is usually internal name, on libvirt host name.

Reply:
Yeah, forgot about the multiple AZ thing.

So, yeah, this is kind of confusing. The *node* object's name is determined by what the kubelet registers. The *machine* object's name is this string copied into `generateName`, so basically this with some extra characters.